src/lib/mapveu/MapVeuAnalysisList.tsx

```diff
@@ -5,32 +5,46 @@ import { safeHtml } from '@veupathdb/wdk-client/lib/Utils/ComponentUtils';
 import { AnalysisClient } from '../core/api/AnalysisClient';
 import { usePromise } from '../core/hooks/promise';
 import { Loading } from '@veupathdb/wdk-client/lib/Components';
+import { createComputation } from '../core/components/computations/Utils';
 
 interface Props {
   studyId: string;
   analysisStore: AnalysisClient;
+  singleAppMode?: string;
 }
 
 export function AnalysisList(props: Props) {
-  const { analysisStore, studyId } = props;
+  const { analysisStore, studyId, singleAppMode } = props;
   const studyRecord = useStudyRecord();
   const list = usePromise(
     useCallback(async () => {
-      const studies = await analysisStore.getAnalyses();
-      return studies.filter((study) => study.analysisId === studyId);
+      const analyses = await analysisStore.getAnalyses();
+      return analyses.filter((analysis) => analysis.studyId === studyId);
     }, [studyId, analysisStore])
   );
   const { url } = useRouteMatch();
   const history = useHistory();
   const createAnalysis = useCallback(async () => {
+    // FIXME Only create computation if singleAppMode is defined
+    // If using singleAppMode, create a computation object that will be used in our default analysis.
+    const computation = singleAppMode
+      ? createComputation(
+          singleAppMode,
+          undefined,
+          [],
+          [],
+          undefined,
+          'Unnamed computation'
+        )
+      : undefined;
     const { analysisId } = await analysisStore.createAnalysis(
-      makeNewAnalysis(studyId)
+      makeNewAnalysis(studyId, computation)
     );
     history.push(`${url}/${analysisId}`);
-  }, [analysisStore, history, studyId, url]);
+  }, [analysisStore, history, singleAppMode, studyId, url]);
   return (
```

Comment:
😊  While you're in here, there's another key error

```js
        <ul>
          {list.value?.map((analysis) => (
            <li> // 👈 key={analysis.analysisId}
              <Link to={`${url}/${analysis.analysisId}`}>
```

https://github.com/VEuPathDB/web-eda/blob/6e264755f0f4e888cae961d22cb8e69e20cc296c/src/lib/mapveu/MapVeuAnalysisList.tsx#L61